core/zoe/docs/zoe.md

```diff
@@ -0,0 +1,263 @@
+I expect *at least* three wood back." [Learn more about the particulars
+of structuring an offer here](TODO). 
+
+Offers are a structured way of describing user intent. To a certain
+extent, an offer description is the user's *contractual understanding*
+of the agreement they are entering into. You might have noticed that
+the offer doesn't specify the mechanism by which the exchange happens.
+The offer doesn't say whether the item you want is up for auction, in
+an exchange, or part of a private trade. The offer doesn't mention the
+particular mechanism because an important part of the design of Zoe is
+a __separation of concerns__. Zoe is responsible for enforcing what we
+call "offer-safety", and the smart contract that runs on top of Zoe is
+responsible for figuring out a proposed reallocation of resources. To
+use an auction as an example, the smart contract is responsible for
+figuring out who wins the auction and how much they pay, but Zoe
+handles the escrowing of the bids and the payments. You can think of
+this as similar to e-commerce websites using a separate
+payment-processor so that they don't have to handle the credit cards
+themselves.
+
+### What is "offer-safety"?
+
+Zoe guarantees offer-safety, meaning that when a user makes an offer
+that is escrowed with Zoe, Zoe guarantees that the user will either
+get what they said they wanted, or the user will get what they
+originally offered back. 
+
+When a user escrows with Zoe, they get two things back immediately: an escrow
+receipt, and a JavaScript promise for a future payout. This escrow
+receipt is what the user can send to smart contracts, as proof that they
+have escrowed the underlying digital assets with Zoe, without the
+smart contract ever having access to the underlying digital assets.
+Let's look a particular example to see how this works.
+
+## An example: A swap
+
+I want to trade my three bricks for five wool. You realize you have
+five wool and agree to the deal. Without Zoe, though, you might send
+me the five wool, and I might disappear without ever giving you the
+three bricks in return. With Zoe, we can safely trade with each
+other, even if we don't trust one another. We can also safely rely on
+the swap smart contract. At worst, if the swap contract behaves badly,
+we will both get a refund, and at best, we'll get what we each wanted.
+
+Let's use the basic `publicSwap` contract ([full text of
+the real contract](/core/zoe/contracts/publicSwap.js)). 
+
+Here's a high-level overview of what would happen:
+1. I make an instance of the swap contract.
+2. I escrow my three bricks with Zoe and get an escrow receipt and a
+   promise for a payout in return.
+3. I send my escrow receipt to the swap as the first offer.
+4. I tell you the swap's `instanceId`
+5. Using the `instanceId`, you look up the swap with Zoe.
+6. You verify that it's using the `publicSwap` contract
+   code you expect, and can ask the swap about the offers made so far.
```

Comment:
"and can ask the swap about the offers made so far". This is an anti-pattern. As we discussed, we can support it. But we should not encourage it.